alonzo/impl/src/Cardano/Ledger/Alonzo/FakePlutus.hs

```diff
@@ -0,0 +1,73 @@
+{-# OPTIONS_GHC -fno-warn-unused-imports #-}
+
+module Cardano.Ledger.Alonzo.FakePlutus where
+
+-- import Plutus.V1.Ledger.Address(Address (..))
+
+import qualified Data.ByteString as BS (ByteString)
+import Language.PlutusTx (Data (..))
+import Language.PlutusTx.IsData.Class (IsData (..))
+import Plutus.V1.Ledger.Ada (adaSymbol, adaToken)
+import Plutus.V1.Ledger.Contexts (TxOutInfo)
+import Plutus.V1.Ledger.Crypto (PubKeyHash (..))
+import Plutus.V1.Ledger.Interval
+  ( Extended (..),
+    Interval (..),
+    LowerBound (..),
+    UpperBound (..),
+  )
+import Plutus.V1.Ledger.Scripts (Datum (..), DatumHash (..), MonetaryPolicyHash (..), ValidatorHash (..))
+import Plutus.V1.Ledger.Slot (SlotRange)
+import Plutus.V1.Ledger.Tx (TxOutRef (..), TxOutType (..))
+import Plutus.V1.Ledger.TxId (TxId (..))
+import Plutus.V1.Ledger.Value (CurrencySymbol (..), TokenName (..), Value (..), singleton, unionWith)
+
+-- ========================================================
+
+-- | Legal addresses may have Staking credential, used to assign rewards
+newtype StakingHash = StakingHash {getStakingHash :: BS.ByteString}
+
+-- | This is isomorphic to the old Plutus stype Address
+data Credential
+  = PubKeyCredential !PubKeyHash
+  | ScriptCredential !ValidatorHash
+
+-- | The new style Plutus Address has two kinds of credentials, normal and staking
+data Address = Address !Credential !(Maybe StakingHash)
+
+-- | The TxInIfo 'resolves' the TxIn from the Tx, using the UTxO and the PtrMap
+data TxInInfo = TxInInfo
+  { txInInfoOutRef :: !TxOutRef,
+    txInAddr :: !Address,
+    txInValue :: !Value,
+    txInDataHash :: !(Maybe DatumHash)
```

Comment:
Why inline the `TxOut` fields instead of just having a field of type `TxOut`? I guess there's a usability/maintainability tradeoff here... my instinct is to go for `data TxInfo = TxInInfo { txInInfoRef :: TxOutRef, txInInfoOut :: TxOut }`

Reply:
I'd also be tempted to have a `Maybe RedeemerHash` here. We need to decide if we're doing the pointer thing or what.

Reply:
We now use TxOut, rather than inlining